applications/StructuralMechanicsApplication/python_scripts/structural_response.py

```diff
@@ -363,4 +374,69 @@ def _SynchronizeAdjointFromPrimal(self):
+                primal_parameters = Parameters( parameter_file.read() )
+
+            # check that HDF5 process is not there
+            if primal_parameters["processes"].Has("list_other_processes"):
+                for i in range(0,primal_parameters["processes"]["list_other_processes"].size()):
+                    process = primal_parameters["processes"]["list_other_processes"][i]
+                    raise Exception("Auto setup of adjoint parameters does not support {} in list_other_processes".format(process["python_module"].GetString()))
+
+            # clone primal settings as base for adjoint
+            adjoint_parameters = primal_parameters.Clone()
+
+            # analysis settings
+            solver_settings = adjoint_parameters["solver_settings"]
+            primal_solver_type = solver_settings["solver_type"].GetString()
+            if primal_solver_type != "static":
+                raise Exception("Auto setup of adjoint parameters does not support {} solver_type. Only available for 'static'".format(primal_solver_type))
+            solver_settings["solver_type"].SetString("adjoint_"+primal_solver_type)
+
+            if not solver_settings.Has("compute_reactions"):
+                solver_settings.AddEmptyValue("compute_reactions")
+            solver_settings["compute_reactions"].SetBool(False)
+
+            if not solver_settings.Has("move_mesh_flag"):
+                solver_settings.AddEmptyValue("move_mesh_flag")
+            solver_settings["move_mesh_flag"].SetBool(False)
+
+            if not solver_settings.Has("scheme_settings"):
+                tmp = solver_settings.AddEmptyValue("scheme_settings")
+                if not tmp.Has("scheme_type"):
+                    tmp.AddEmptyValue("scheme_type")
+            solver_settings["scheme_settings"]["scheme_type"].SetString("adjoint_structural")
+
+            # Dirichlet conditions: change variables
+            for i in range(0,primal_parameters["processes"]["constraints_process_list"].size()):
+                process = adjoint_parameters["processes"]["constraints_process_list"][i]
+                variable_name = process["Parameters"]["variable_name"].GetString()
+            process["Parameters"]["variable_name"].SetString("ADJOINT_"+variable_name)
+
+            # Neumann conditions - do not modify to read the same load values as in primal:
+
+            # Output process:
+            # TODO how to add the output process? How find out about the variables?
+            if adjoint_parameters.Has("output_configuration"):
+                Logger.PrintInfo("> Output process is removed for adjoint analysis. To enable it define adjoint_parameters yourself.")
+                adjoint_parameters.RemoveValue("output_configuration")
+
+            # sensitivity settings
+            adjoint_parameters["solver_settings"].AddValue("sensitivity_settings", self.response_settings["sensitivity_settings"])
+            self.response_settings.RemoveValue("sensitivity_settings")
```

Comment:
Why do you remove the sensitivity settings here?

Reply:
because technically they are not part of the settings that are passed down to the cpp response.
The sensitivity_settings are only needed in the sensitivty builder a.k.a adjoint postprocess.

Reply:
Still, it seems like not a very good practice here. One would expect that a single settings object always just adds settings but never removes, otherwise, why should it be there in the first place, if it is just needed locally. But I do not have the complete picture here, so there might be indeed good reasons to do so.

Reply:
Maybe we discuss it in person.